packages/core/src/modules/vc/jwt-vc/W3cJwtCredentialService.ts

```diff
@@ -0,0 +1,530 @@
+   *    an error is thrown.
+   *
+   * The signer (`iss`) of the JWT is verified against the `controller` of the verificationMethod resolved in the did
+   * document. This means if the `iss` of a credential is `did:example:123` and the controller of the verificationMethod
+   * is `did:example:456`, an error is thrown to prevent the JWT from successfully being verified.
+   *
+   * In addition the JWT must conform to one of the following rules:
+   *   - MUST be a credential and have an `iss` field and MAY have an absolute or relative `kid`
+   *   - MUST not be a credential AND ONE of the following:
+   *      - have an `iss` field and MAY have an absolute or relative `kid`
+   *      - does not have an `iss` field and MUST have an absolute `kid`
+   */
+  private async getVerificationMethodForJwtCredential(
+    agentContext: AgentContext,
+    options: {
+      credential: W3cJwtVerifiableCredential | W3cJwtVerifiablePresentation
+      purpose?: DidPurpose[]
+    }
+  ) {
+    const { credential, purpose } = options
+    const kid = credential.jwt.header.kid
+
+    const didResolver = agentContext.dependencyManager.resolve(DidResolverService)
+
+    // The signerId is the `holder` of the presentation or the `issuer` of the credential
+    // For a credential only the `iss` COULD be enough to resolve the signer key (see method comments)
+    const signerId = credential.jwt.payload.iss
+
+    let verificationMethod: VerificationMethod
+
+    // If the kid starts with # we assume it is a relative did url, and we resolve it based on the `iss` and the `kid`
+    if (kid?.startsWith('#')) {
+      if (!signerId) {
+        throw new AriesFrameworkError(`JWT 'kid' MUST be absolute when when no 'iss' is present in JWT payload`)
+      }
+
+      const didDocument = await didResolver.resolveDidDocument(agentContext, signerId)
+      verificationMethod = didDocument.dereferenceKey(`${signerId}${kid}`, purpose)
+    }
+    // this is a full did url (todo check if it contains a #)
+    else if (kid && isDid(kid)) {
+      const didDocument = await didResolver.resolveDidDocument(agentContext, kid)
+
+      verificationMethod = didDocument.dereferenceKey(kid, purpose)
+
+      if (signerId && didDocument.id !== signerId) {
+        throw new AriesFrameworkError(`kid '${kid}' does not match id of signer (holder/issuer) '${signerId}'`)
+      }
+    } else {
+      if (!signerId) {
+        throw new AriesFrameworkError(`JWT 'iss' MUST be present in payload when no 'kid' is specified`)
+      }
+
+      // Find the verificationMethod in the did document based on the alg and proofPurpose
+      const jwkClass = getJwkClassFromJwaSignatureAlgorithm(credential.jwt.header.alg)
+      if (!jwkClass) throw new AriesFrameworkError(`Unsupported JWT alg '${credential.jwt.header.alg}'`)
+      const { supportedVerificationMethodTypes } = getKeyDidMappingByKeyType(jwkClass.keyType)
```

Comment:
```suggestion
      if (!jwkClass) throw new AriesFrameworkError(`Unsupported JWT alg '${credential.jwt.header.alg}'`)
      
      const { supportedVerificationMethodTypes } = getKeyDidMappingByKeyType(jwkClass.keyType)
```

Reply:
It seems there's not changed here except for the newline? Is that what you meant?